packages/venia-ui/lib/util/images.js

```diff
@@ -17,23 +17,48 @@ export const imageWidths = {
     XXXLARGE: 2560
 };
 
-const generateURL = (imageURL, mediaBase) => (width, height) => {
-    const url = resourceUrl(imageURL, {
+const generateURL = (imageURL, mediaBase) => (width, height) =>
+    resourceUrl(imageURL, {
         type: mediaBase,
         width,
         height
     });
 
-    return `${url} ${width}w`;
+export const generateURLFromContainerWidth = (
+    imageURL,
+    type = 'image-product',
+    containerWidth
```

Comment:
> Consider making the signature of this function take an "options" object with each of these properties to improve the readability of the call site and so that callers don't have to care about what order they're in.

Makes sense. Will do.

Reply:
> How do images know their `containerWidth`? Are we expecting them to do DOM calculations?

No, this is expected to be present with the component. For instance, the product details page already knows where the image will be placed and how big the image container will be. 

Come to think of it, I am doubting if the component will know the width in all cases.

I am thinking of DOM calculations but I am sure there might be a better way to do this that I am missing. Any idea how should the component get the width if not provided?